File: app/src/main/java/org/tvheadend/tvhclient/ui/features/playback/external/ConvertHostnameToAddressTask.java
package org.tvheadend.tvhclient.ui.features.playback.external;

import android.os.AsyncTask;
import androidx.annotation.NonNull;

import org.tvheadend.tvhclient.domain.entity.Connection;

import java.net.InetAddress;
import java.net.UnknownHostException;

import timber.log.Timber;

class ConvertHostnameToAddressTask extends AsyncTask<Void, Void, String> {

    private final Connection connection;

    ConvertHostnameToAddressTask(@NonNull Connection connection) {
        this.connection = connection;
    }

    @Override
    protected String doInBackground(Void... voids) {
        try {
            return  InetAddress.getByName(connection.getHostname()).getHostAddress();
        } catch (UnknownHostException e) {
            Timber.d("Could not get ip address from " + connection.getHostname() + ", using hostname as fallback", e);
            return connection.getHostname();
        }
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/ui/features/playback/external/ConvertHostnameToAddressTask.kt
package org.tvheadend.tvhclient.ui.features.playback.external

import kotlinx.coroutines.CoroutineScope
import org.tvheadend.tvhclient.util.extensions.executeAsyncTask
import timber.log.Timber
import java.net.InetAddress
import java.net.UnknownHostException

class ConvertHostnameToAddressTask(lifecycleScope: CoroutineScope, private val hostname: String) {

    private var convertedHostname = hostname
    init {
        lifecycleScope.executeAsyncTask(onPreExecute = {
            // ... runs in Main Thread
            Timber.d("onPreExecute")
        }, doInBackground = {
            Timber.d("doInBackground")
            try {
                convertedHostname = InetAddress.getByName(hostname).hostAddress
            } catch (e: UnknownHostException) {
                Timber.d(e, "Could not get ip address from $hostname, using hostname as fallback")
            }
        }, onPostExecute = {
            // runs in Main Thread
            Timber.d("onPostExecute")
            convertedHostname
        })
    }
}
